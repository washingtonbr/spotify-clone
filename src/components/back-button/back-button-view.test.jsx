import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import BackButtonView from './back-button-view';

const mockGoBack = jest.fn();

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn().mockImplementation(() => ({
    goBack: mockGoBack,
  })),
}));

describe('BackButtonView', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<BackButtonView />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should navigate to previus page', () => {
    const wrapper = mount(<BackButtonView />);

    wrapper
      .find('button')
      .first()
      .simulate('click');

    expect(mockGoBack).toBeCalledTimes(1);
  });
});
