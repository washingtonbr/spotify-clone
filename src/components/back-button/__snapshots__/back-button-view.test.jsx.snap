// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButtonView render corectly 1`] = `
<button
  className="sc-AxjAm cOdxoW"
  onClick={[MockFunction]}
>
  <i
    className="sc-AxirZ iyWABq"
  />
  Voltar
</button>
`;
